GiniVision/Classes/PreferredResource.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  PreferredResource.swift
+//  Pods
+//
+//  Created by Nikola Sobadjiev on 16/03/2017.
+//
+//
+
+import UIKit
+
+enum ResourceSource {
+    case unknown
+    case library
+    case custom
+}
+
+/*
+ * PreferredResource typically controls what artwork (image and text) would be used in a UI element
+ * such as a button or a bar button item. Resources in the Gini Vision Library are usually brandable
+ * via UIAppearance, Asset catalogs and localizable strings. For instance, many buttons have a default
+ * image in the library's Asset catalog, but clients are free to add an image with the same image in
+ * their catalog. In this case, the customized image will be used.
+ */
+
+struct PreferredResource {
+    
+    let imageName:String?
+    let text:String?
+    let textComment:String?
+    let appBundle = Bundle.main
+    let libBundle = Bundle(for: GiniVision.self)
+    var imageSource:ResourceSource {
+        if let name = imageName {
+            if UIImage(named: name, in: appBundle, compatibleWith: nil) != nil {
```

Comment:
I couldn't make it happen unfortunately. I thought I could check if there is such file in one bundle or the other. But since we use Asset catalogs, that doesn't work. Maybe I can research a bit more, but so far it seems there's no way to check for files inside an Asset catalog. I think it's because they compile into a simple ".car".

I'm not terribly concerned about the performance of the code thought. After an UIImage is loaded from the file system, it's automatically cached, so any subsequent request for it will have constant complexity. Alternatively, if it's not there, I assume there's hardly any file operation, so that would be really fast as well.